### PR TITLE
Add an error in case a name field has content of the pattern field

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
It's a common issue when edition the gir files so instead of silently fail, it now displays an error.

cc @sdroege

### CODE REVIEW

File: src/config/ident.rs [hunk 1]
```diff
@@ -46,10 +46,20 @@ impl Ident {
                     e
                 })
                 .ok(),
-            None => toml
-                .lookup("name")
-                .and_then(Value::as_str)
-                .map(|s| Ident::Name(s.into())),
+            None => match toml.lookup("name").and_then(Value::as_str) {
+                Some(name) => {
+                    if name.chars().any(|c| c == '.' || c == '+' || c == '*') {
```
Comment (sdroege) [inline src/config/ident.rs:51]:
```suggestion
                    if name.contains(&['.', '+', '*']) {
```

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise, but CI doesn't like it for some reason

File: src/config/functions.rs [hunk 2]
```diff
@@ -565,12 +565,13 @@ pattern = 'func\d+'
 "#,
         );
         let fns = Functions::parse(Some(&toml), "a");
-        assert_eq!(fns.len(), 4);
```
Comment (sdroege) [inline src/config/functions.rs:568]: Why did this change?

File: src/config/properties.rs [hunk 3]
```diff
@@ -184,12 +184,13 @@ pattern = 'prop\d+'
 "#,
         );
         let props = Properties::parse(Some(&toml), "a");
-        assert_eq!(props.len(), 4);
```
Comment (sdroege) [inline src/config/properties.rs:187]: Why did this change?

File: src/config/ident.rs [hunk 4]
```diff
@@ -46,10 +46,23 @@ impl Ident {
...
                 })
                 .ok(),
-            None => toml
-                .lookup("name")
-                .and_then(Value::as_str)
-                .map(|s| Ident::Name(s.into())),
+            None => match toml.lookup("name").and_then(Value::as_str) {
+                Some(name) => {
+                    // we're forced to do it in two steps otherwise we get a nice compiler error:
+                    // expected a `std::ops::Fn<(char,)>` closure, found `[char; 3]`
```
Comment (sdroege) [inline src/config/ident.rs:52]: Not really forced, that's your decision :) The other options are all shorter!

File: src/config/functions.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/config/functions.rs:568]: The explanation is in the comment just below. ;)

File: src/config/properties.rs (hunk 3 above)
Comment (GuillaumeGomez) [reply src/config/properties.rs:187]: The explanation is in the comment just below. ;)

Comment (sdroege) [reply src/config/properties.rs:187]: Which comment?

File: src/config/functions.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/config/functions.rs:568]: This one: https://github.com/gtk-rs/gir/pull/1027/files#diff-ee42c4f4c96eb712946d909affc0319ee2aa8c12db41446d441a5144191269e4R573 :)

Comment (sdroege) [reply src/config/functions.rs:568]: Oh the test was broken before...

**Review by sdroege — APPROVED** [review]

**Pull request merged**